internal/appsec/config.go

```diff
@@ -69,9 +70,25 @@ func readWAFTimeoutConfig() (timeout time.Duration) {
 	if value == "" {
 		return
 	}
+
+	// Check if the value contains any letter, which means the user has
+	// specified its own time duration unit(s) such as 1s200ms30ns
+	hasLetter := false
+	for _, r := range value {
+		if unicode.IsLetter(r) {
+			hasLetter = true
+			break
+		}
+	}
+	// Add the default microsecond time-duration suffix if no letter was found
+	if !hasLetter {
+		value += "us"
+	}
```

Comment:
I'm not sure why iterating over the whole string is necessary. Can't we just check for last char?
We know `value` is not empty, so there are really only two cases I can think of:
- `value` is of the form `123s`, in which case nothing happens
- `value` is of the form `123`, in which case we add a suffix

Other forms such as `1s2ms3us` are just the same cases with a different prefix.
Other cases where `value` is ill formed will fail through the call to `time.ParseDuration`.
WDYT?
```suggestion
        // Add the default microsecond time-duration suffix if no trailing letter was found
	if !unicode.IsLetter(rune(s[len(s)-1])) {
                s += "us"
        }
```